internal/ingest/upload_test.go

```diff
@@ -145,13 +151,89 @@ func TestUpload(t *testing.T) {
 		},
 		{
 			name: "Uploads a SIP",
+			claims: &auth.Claims{
+				ISS: "http://keycloak:7470/realms/artefactual",
+				Sub: "1234567890",
+			},
 			mock: func(ctx context.Context, psvc *persistence_fake.MockService, tc *temporalsdk_mocks.Client) {
+				psvc.EXPECT().ReadJWTUser(
+					mockutil.Context(),
+					"http://keycloak:7470/realms/artefactual",
+					"1234567890",
+				).Return(&datatypes.User{
+					UUID:   uuid1,
+					Name:   ref.New("Test User"),
+					Email:  ref.New("nobody@example.com"),
+					JWTIss: ref.New("http://keycloak:7470/realms/artefactual"),
+					JWTSub: ref.New("1234567890"),
+				}, nil)
+
 				psvc.EXPECT().CreateSIP(
-					ctx,
+					mockutil.Context(),
 					&datatypes.SIP{
-						UUID:   uuid0,
-						Name:   "first.zip",
-						Status: enums.SIPStatusQueued,
+						UUID:       uuid0,
+						Name:       "first.zip",
+						Status:     enums.SIPStatusQueued,
+						UploaderID: &uuid1,
+					},
+				).Return(nil)
+
+				tc.On(
+					"ExecuteWorkflow",
+					mock.AnythingOfType("*context.timerCtx"),
+					temporalsdk_client.StartWorkflowOptions{
+						ID:                    fmt.Sprintf("processing-workflow-%s", uuid0.String()),
+						TaskQueue:             "test",
+						WorkflowIDReusePolicy: temporalsdk_api_enums.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE,
+					},
+					ingest.ProcessingWorkflowName,
+					&ingest.ProcessingWorkflowRequest{
+						SIPUUID:   uuid0,
+						SIPName:   "first.zip",
+						Type:      enums.WorkflowTypeCreateAip,
+						Key:       key,
+						Extension: ".zip",
+					},
+				).Return(nil, nil)
+			},
+			multipartBody: zipMmultipartBody,
+			contentType:   "multipart/form-data; boundary=foobar",
+			maxUploadSize: 102400000,
+			want:          &goaingest.UploadSipResult{UUID: uuid0.String()},
+		},
+		{
+			name: "Uploads a SIP and creates a user",
```

Comment:
It would be nice to add another success test case, where authentication is disabled and it creates a SIP without uploader.